src/gui/inputdevices/HandRecognition.cpp

```diff
@@ -155,14 +158,19 @@ void HandRecognition::penEvent()
  */
 void HandRecognition::enableTouch()
 {
-	if (inputContext)
-	{
-		inputContext->unblockDevice(InputContext::TOUCHSCREEN);
-	}
-	if (touchImpl)
+	if (enabled)
 	{
-		touchImpl->enableTouch();
+		if (inputContext)
+		{
+			inputContext->unblockDevice(InputContext::TOUCHSCREEN);
+		}
+		if (touchImpl)
+		{
+			touchImpl->enableTouch();
+		}
 	}
+	
+	inputContext->touchHandler->reset();
```

Comment:
I'm not really happy with this, as it is somewhere where it essentially shouldn't be. It should either be called in `InputContext::unblockDevice()` or even within the handler itself by overriding `AbstractInputHandler::block()`.

What you currently do is violating a few common programing paradigms. Namely the encapsulation paradigm. And violating this one leads to huuuge problems later on.

Reply:
Think about calling AbstractInputHandler::block() within the method when you override it. Otherwise you will break something else.